FWCore/Services/plugins/InitRootHandlers.cc

```diff
@@ -346,7 +377,55 @@ namespace {
       return full_write(2, text);
     }
 
+    void sig_pause_for_stacktrace(int sig, siginfo_t*, void*) {
+      char buff[128] = "\nModule: ";
+ 
+      if (edm::CurrentModuleOnThread::getCurrentModuleOnThread() != nullptr) {
+        strlcat(buff, edm::CurrentModuleOnThread::getCurrentModuleOnThread()->moduleDescription()->moduleName().c_str(), 128);
+        full_cerr_write(buff);
+      }
+
+      struct timespec timeout = { edm::service::InitRootHandlers::stackTracePause(), 0 };
+#if defined(__linux__)
+      sigset_t sigset;
+      sigemptyset(&sigset);
+      sigaddset(&sigset, SIGCONT);
+      sigtimedwait(&sigset, NULL, &timeout);
+#else
+      nanosleep(&timeout); // macOS doesn't have sigtimedwait()
+#endif
+    }
+
     void sig_dostack_then_abort(int sig, siginfo_t*, void*) {
+      char buff[128] = "\nModule: ";
+
+      auto&& tids = edm::service::InitRootHandlers::threadIDs();
```

Comment:
> On Aug 9, 2016, at 11:19, Brian Bockelman <notifications@github.com> wrote:
> 
> In FWCore/Services/plugins/InitRootHandlers.cc:
> 
> >      void sig_dostack_then_abort(int sig, siginfo_t*, void*) {
> > +      char buff[128] = "\nModule: ";
> > +
> > +      auto&& tids = edm::service::InitRootHandlers::threadIDs();
> 
> I forget my C++11 syntax - is this a move? If so, is it guaranteed to avoid memory allocations?

It's a universal reference, in this case it should be equivalent to

const auto& tids = edm::service::InitRootHandlers::threadIDs();

-dan